### ISSUE TITLE
`prefer-await`: incorrectly flags .then() inside class constructors

### ISSUE THREAD

**Issue by rentalhost** [sindresorhus/eslint-plugin-unicorn#3564]
The `prefer-await` rule suggests replacing `.then()` with `await`. However, class constructors cannot be declared as `async`. When a developer needs to handle a Promise directly inside the constructor (e.g., wrapping a passed Promise to a class property so internal logic runs before the instance is consumed), they are forced to use `.then()`. The rule should ignore `.then()` usage when it is located inside a class constructor body.

```ts
class Loader {
  private readonly asset: Promise<Asset>;

  public constructor(source: Promise<Source>) {
    this.asset = source.then((raw) => {
      raw.prepare();

      return raw;
    });
  }

  public async use() {
    const asset = await this.asset;

    return asset.render();
  }
}
```

**Comment by sindresorhus** [comment]
You can use async IIFE: https://github.com/sindresorhus/eslint-plugin-unicorn/commit/e2a834911632550fa5aa7bc9bea7637c07691df1
